packages/gatsby-plugin-offline/src/gatsby-node.js

```diff
@@ -0,0 +1,50 @@
+const Promise = require(`bluebird`)
+const precache = require(`sw-precache`)
+const path = require(`path`)
+
+exports.createPages = () => [
+  {
+    path: `/offline-plugin-app-shell-fallback/`,
+    component: path.resolve(`${__dirname}/app-shell.js`),
+  },
+]
+
+exports.postBuild = () => new Promise((resolve, reject) => {
+  const rootDir = `public`
+
+  const options = {
+    staticFileGlobs: [
+      `${rootDir}/**/*.{js,woff2}`,
+      `${rootDir}/index.html`,
+      `${rootDir}/manifest.json`,
+      `${rootDir}/offline-plugin-app-shell-fallback/index.html`,
+    ],
+    stripPrefix: rootDir,
+    navigateFallback: `/offline-plugin-app-shell-fallback/index.html`,
+    cacheId: `gatsby-plugin-offline`,
+    dontCacheBustUrlsMatching: /(.*.woff2|.*.js)/,
```

Comment:
It's probably safer to include a RegExp here that matches the specific hash format you're using in your versioned filenames. So if you're using an 8 character hash surrounded by `.` characters, leading to filenames like 'app.123456ab.js`, something like `/\.\w{8}\./` would make sense.

You don't need to match the entire URL, so there's no need for the leading `.*` wildcard.